### PR TITLE
No symbols computed if max symbols is zero

### LINKED ISSUE #898: Symbols Max Items Computed doesn't work for 0
If `xml.symbols.maxItemsComputed` is set to 0, then all the symbols are still computed.

### PR DESCRIPTION
**Pull request by datho7561**
Closes #898

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

**Comment by datho7561** [comment]
Unit test works, but in practice, this doesn't actually work.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLDocumentSymbolsTest.java [hunk 1]
```diff
@@ -143,4 +143,21 @@ public void exceedSymbolLimit() {
...
+				"    <c>\n" + //
+				"      <d />\n" + //
+				"    </c>\n" + //
+				"  </b>" + //
+				"</a>";
+
+		XMLSymbolSettings settings = new XMLSymbolSettings();
+		settings.setMaxItemsComputed(0);
+
+		testDocumentSymbolsFor(xml, "test.xml", settings);
```
Comment (fbricon) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLDocumentSymbolsTest.java:160]: what is the assertion here?

Comment (datho7561) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLDocumentSymbolsTest.java:160]: These are no document symbols when the limit is zero. I will make this test a bit more clear.

Comment (datho7561) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLDocumentSymbolsTest.java]: Please let me know if this clarifies the test

**Comment by datho7561** [comment]
[ ] Note to self: address comment on reference to this PR

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolSettings.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 	private String[] excluded;
 
-	private int maxItemsComputed;
+	private int maxItemsComputed = 5000;
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolSettings.java:27]: Define a private constant like 

`private static final DEFAULT_MAX_ITEMS_COMPUTED = 5000;`

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/XMLAssert.java [hunk 3]
```diff
@@ -766,6 +766,33 @@ public static void testDocumentSymbolsFor(XMLLanguageService xmlLanguageService,
 
 	}
 
+	public static void testDocumentSymbolsNumber(String xml, String fileURI, XMLSymbolSettings symbolSettings,
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/XMLAssert.java:769]: XMLAssert is a kind of framework to test the XML language services features.

As this XMLAssert class is used for any extension, I would like to provide only services test which are a lot of used (ex : test for completion, hover, validation, etc). 

I'm not sure that a lot extension will do some test with testDocumentSymbolsNumber. So please move those methods in XMLDocumentSymbolsTest.java. 

We could change that according user issues.

**Comment by angelozerr** [comment]
Please write a comment in vscode-xml  / docs / Symbols about this 0 value. Thanks!

**Pull request merged**